src/nvim/api/command.c

```diff
@@ -1086,8 +1086,9 @@ void create_user_command(String name, Object command, Dict(user_command) *opts,
     def = opts->range.data.integer;
     addr_type_arg = ADDR_LINES;
   } else if (HAS_KEY(opts->range)) {
-    api_set_error(err, kErrorTypeValidation, "Invalid value for 'range'");
-    goto err;
+    VALIDATE_S(false, "range", "", {
+      goto err;
```

Comment:
If we had a consistent `validation_fail` label in the API impls we could omit this extra boilerplate in all the `VALIDATE` calls, though it would make control flow less obvious.

Reply:
If anything I'd rather search for a route that eliminates the use of `goto`.